### PR TITLE
Replace instanceof with check for internal object type

### PR DESCRIPTION
**Pull request by nbermudezs**
Relates to ongoing conversation in https://github.com/Yomguithereal/baobab-react/issues/123

### CODE REVIEW

File: src/higher-order.js [hunk 1]
```diff
@@ -30,7 +30,7 @@ function invalidMapping(name, mapping) {
  * Root component
  */
 function root(tree, Component) {
-  if (!(tree instanceof Baobab))
+  if (!tree || tree.toString() !== '[object Baobab]')
```
Comment (Yomguithereal) [inline src/higher-order.js]: You should probably test for `typeof tree === 'object'` also.

**Comment by Yomguithereal** [comment]
Can you also see if the proptypes must be changed? And can you update the mixin's code also please?

**Comment by nbermudezs** [comment]
Hey @Yomguithereal, sorry I dropped the ball on this, busy couple of weeks. I'll make the suggested changes.

**Comment by nbermudezs** [comment]
Done. Let me know what you think.

File: src/utils/helpers.js [hunk 2]
```diff
@@ -30,3 +30,13 @@ export function solveMapping(mapping, props, context) {
...
 }
+
+/**
+ * Determines if the given tree is a Baobab tree.
+ * FIXME: if Baobab ever implements something like Array.isArray we should use
+ * that instead of relying in the internal _identity = '[object Baobab]' value.
+ * See https://github.com/Yomguithereal/baobab/blob/master/src/baobab.js#L111
+ */
+export function isBaobabTree(tree) {
+    return !!(tree && tree.toString() === '[object Baobab]');
```
Comment (Yomguithereal) [inline src/utils/helpers.js]: If you want to be paranoid, you should also check that `tree.toString` is a function.

Comment (nbermudezs) [inline src/utils/helpers.js]: Absolutely! I like being paranoid 😬

**Comment by nbermudezs** [comment]
@Yomguithereal let me know if it is good now :)

**Comment by Yomguithereal** [comment]
Sorry for the delay @nbermudezs. Merging and releasing asap.

**Pull request merged**